CHANGELOG.md

```diff
@@ -2,6 +2,122 @@
 
 This contains only the most important and/or user-facing changes; for a full changelog, see the commit history.
 
+## [2.0.0](https://github.com/ably/ably-js/tree/2.0.0) (date TBD)
+
+The 2.0.0 release introduces a number of new features and QoL improvements, including a new way to remove bloat and reduce the bundle size of your ably-js client, first-class support for Promises, a more idiomatic approach to using ably-js' React Hooks, enhancements to TypeScript typings, and more.
+
+Below is a list of all new features, APIs, deprecations, and breaking changes.
+Read [ably-js v2 migration guide](link TBD) for more information.
+
+### Bundle Size Reduction
+
+The default bundle size for the web has been reduced by 30% compared to v1 - from 231.35 KiB to 162.15 KiB. When calculated with gzip compression, the reduction is 28.5%, from 81.68 KiB to 58.48 KiB.
+
+Additionally, by utilizing the new modular variant of the library and JavaScript tree shaking, you can create your own minimal useful `Realtime` client and achieve a bundle size reduction of 60% compared to v1 - from 231.35 KiB to 93.29 KiB (or 65% for gzip: from 81.68 KiB to 28.29 KiB).
+
+### Modular variant of the library
+
+An ESM variant of the library is now available for browsers (but not for Node.js) at `ably/modular`. This modular variant of the library supports tree shaking, allowing for a reduction in the Ably bundle size within your application. It can also be used by Web Workers.
+
+### First-class support for Promises
+
+The exports `ably/callbacks` and `ably/promises` have been removed. The callbacks API has been entirely removed, and now only promises are supported. Calls like `new Ably.Realtime.Promise` should be replaced with `new Ably.Realtime`.
```

Comment:
Bearing in mind your approach described [above](https://github.com/ably/ably-js/pull/1656/files#r1529432940):

> The changelog lists all the things that were added, changed, and improved, while the migration guide explains step by step how to migrate.

…it probably would be best not to include this instruction, since it's not a very useful one in isolation and requires the migration guide to properly understand how a user needs to change their code. What do you think?

Reply:
Aggree, reworded to describe the change more generally and don't include specific instructions